modules_core/com.smf.jobs.defaults/src/com/smf/jobs/defaults/invoices/CreateFromOrder.java

```diff
@@ -108,16 +118,58 @@ protected ActionResult action(JSONObject parameters, MutableBoolean isStopped) {
       OBDal.getInstance().rollbackAndClose();
     }
 
+    ResponseActionsBuilder.MessageType messageType;
+    if (result.getType() == Result.Type.ERROR) {
+      messageType = ResponseActionsBuilder.MessageType.ERROR;
+    } else {
+      messageType = ResponseActionsBuilder.MessageType.SUCCESS;
+    }
+
     result.setResponseActionsBuilder(
         getResponseBuilder()
             .retryExecution()
             .refreshGridParameter(ORDER_GRID_PARAM)
-            .showMsgInProcessView(ResponseActionsBuilder.MessageType.SUCCESS, result.getMessage())
+            .showMsgInProcessView(messageType, result.getMessage())
     );
 
     return result;
   }
 
+  /**
+   * Checks if the BusinessPartner associated with an order specified in the given parameters has an invoice address.
+   *
+   * @param parameters
+   *     A JSONObject containing order details, including an order ID.
+   * @return true if the BusinessPartner associated with the order has an invoice address; false otherwise.
+   * @throws JSONException
+   *     If there is an error parsing the JSON data.
+   */
+  private boolean checkBusinessPartnerInvoiceAddress(JSONObject parameters) throws JSONException {
```

Comment:
Maybe this method is unnecessary because the invoice address of an order is mandatory, then all orders must have invoice address.

Reply:
That's true, but since the order was created from the 'Sales Quotation' window instead of 'Sales Order', it assigns an address. However, the Business Partner doesn't have a billing address configured, which leads to an issue when reaching the 'Create Invoice from Order' window. Do you think I should focus on the process of creating the order from 'Sales Quotation' instead?